### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by Bread2018** [open-mmlab/mmdetection#9247]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

sys.platform: linux
Python: 3.8.10 (default, Jun  4 2021, 15:09:15) [GCC 7.5.0]
CUDA available: True
GPU 0: NVIDIA GeForce RTX 2060
CUDA_HOME: None
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 1.12.1
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.6
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.6, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.13.1
OpenCV: 4.6.0
MMCV: 1.7.0
MMCV Compiler: GCC 9.3
MMCV CUDA Compiler: 11.6
MMDetection: 2.25.2+9d3e162
 
ROS2 foxy
Realsense-ros
opencv

How you installed PyTorch: conda
Other environment variables that may be related: 
PATH:
/home/bread/anaconda3/envs/pytorch-test/bin:/home/bread/ros2_ws/install/costmap_converter/bin:/home/bread/autodock_ros2_ws/install/apriltag_ros/bin:/home/bread/autodock_ros2_ws/install/apriltag/bin:/opt/ros/foxy/bin:/home/bread/anaconda3/condabin:/home/bread/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

PYTHONPATH:
/home/bread/ros2_ws/install/teb_msgs/lib/python3.8/site-packages:/home/bread/ros2_ws/install/ros2_aruco_interfaces/lib/python3.8/site-packages:/home/bread/ros2_ws/install/ros2_aruco/lib/python3.8/site-packages:/home/bread/ros2_ws/install/robot_vision_ros2/lib/python3.8/site-packages:/home/bread/ros2_ws/install/robot_navigation_ros2/lib/python3.8/site-packages:/home/bread/ros2_ws/install/robot_description/lib/python3.8/site-packages:/home/bread/ros2_ws/install/realsense2_camera_msgs/lib/python3.8/site-packages:/home/bread/ros2_ws/install/costmap_converter_msgs/lib/python3.8/site-packages:/home/bread/ros2_ws/install/base_control_ros2/lib/python3.8/site-packages:/home/bread/autodock_ros2_ws/install/autodock_msgs/lib/python3.8/site-packages:/home/bread/autodock_ros2_ws/install/apriltag_msgs/lib/python3.8/site-packages:/home/bread/dev_ws/install/two_wheeled_robot/lib/python3.9/site-packages:/opt/ros/foxy/lib/python3.8/site-packages

### Reproduces the problem - code sample

#after i launch realsense camera node (ros2 launch realsense2_camera rs_launch.py)
#then i launch another python file whitch subscribe color stream topic for finding object bounding box, and the import down below #is in it. 
from mmdet.models import build_detector

import pyrealsense2 as rs
import numpy as np
import cv2
import mmcv
from mmcv.runner import load_checkpoint
from rcl_interfaces.msg import ParameterDescriptor
from geometry_msgs.msg import PoseStamped,Twist 
from mmdet.apis import inference_detector, show_result_pyplot
from mmdet.models import build_detector
from std_msgs.msg import String
from sensor_msgs.msg import Image, CameraInfo
import message_filters
import cv2
from rclpy.node import Node
from cv_bridge import CvBridge, CvBridgeError
import rclpy
class FrameListener(Node):
    def __init__(self):
        super().__init__('get_image')

### Load model
        config = '/home/bread/mmdetection/configs/pallet/yolact_r50_1x8_pallet_20220711.py'
        # Setup a checkpoint file to load
        checkpoint = '/home/bread/mmdetection/work_dir/latest.pth'
        # Set the device to be used for evaluation
        self.device='cuda:0'

        # Load the config
        config = mmcv.Config.fromfile(config)
        # Set pretrained to be None since we do not need pretrained model here
        config.model.pretrained = None

        # Initialize the detector
        self.model = build_detector(config.model)

        # Load checkpoint
        checkpoint = load_checkpoint(self.model, checkpoint, map_location=self.device)

        # Set the classes of models for inference
        self.model.CLASSES = checkpoint['meta']['CLASSES']

        # We need to set the model's cfg for inference
        self.model.cfg = config

        # Convert the model to GPU
        self.model.to(self.device)
        # Convert the model into evaluation mode
        self.model.eval()

### Reproduces the problem - command or script

ros2 launch realsense2_camera rs_launch.py
python3 DLnodetest.py (for finding center point of bounding box)

### Reproduces the problem - error message

free(): invalid pointer
Aborted (core dumped)

### Additional information

I have a I python file for finding center point from detected bounding box without ros parameter，when i run this file alone, it works well, But when i run the detection node after the realsense node is running(ros2 launch realsense2_camera rs_launch.py), the error will occurd.

**Comment by ZwwWayne** [comment]
Hi @Bread2018 ,
Thanks for your kind PR. Would you like to provide a more detailed error message? For now, we cannot find any useful information to debug it. the core dumped can be caused by many reasons.

**Comment by Bread2018** [comment]
I have solved this problem, it's not caused by mmdetection, it's my code problem, sorry for bothering you, thank you.
